src/interpreter/kernel/exec.js

```diff
@@ -1089,6 +1089,15 @@ export function executeStackFrame(frame: StackFrame, depth: number = 0): any {
 
       break;
     }
+
+    case 'reinterpret/i64': {
+      
+      const c1 = pop1('i64');
+      const res = f64.createValue(c1.value.toNumber());
```

Comment:
There are probably some cases where `c1.value.toNumber()` returns an invalid value?

Note that currently `pushResult` can push `undefined` on the stack (I have duplicated code all over the kernel/exec to handle this case, we could maybe include it directly in pushResult).

Reply:
I don't think that `Long.toNumber` does the same thing as `reinterpret/i64`. Long performs rounding -> changes bits to get the most similar value. Reinterpretation does not change the bits. In this case it should take upper and lower stored by Long and interpret these 64 bits as IEEE binary64 floating point number.

* [**Long.toNumber**](https://github.com/dcodeIO/long.js#longtonumber)
* [**reinterpret/i64**](https://webassembly.github.io/spec/core/exec/numerics.html#op-reinterpret)

Reply:
Exactly, this was a hack just to get something testable out the other end - it is not a reinterpret cast.

The key things I wanted to discuss via this PR are the use of long.js and the AST format. Many of the other details are ‘hacked’.

I’d be happy to iterate either and create a more appropriate PR if the overall approach is ok?

Reply:
Okay nice! Currently we don't have the necessary `NaN` payload support in place to get this completely right anyways (#45). I think Long.js will be an overhead in the long ( 😁  ) run, but its definitely a good starting point as discussed in #22, so Im happy with it.